pkg/pillar/dpcreconciler/genericitems/dhcpcd.go

```diff
@@ -147,10 +147,14 @@ func (c *DhcpcdConfigurator) Create(ctx context.Context, item depgraph.Item) err
 				args = append(args, "--static",
 					fmt.Sprintf("routers=%s", config.Gateway.String()))
 			}
-			// XXX do we need to calculate a list for option?
+			var dnsServers []string
 			for _, dns := range config.DnsServers {
+				dnsServers = append(dnsServers, dns.String())
+			}
+			if len(dnsServers) > 0 {
 				args = append(args, "--static",
-					fmt.Sprintf("domain_name_servers=%s", dns.String()))
+					fmt.Sprintf("domain_name_servers=%s",
+						strings.Join(dnsServers, " ")))
```

Comment:
I see this being documented as the syntax for dhcpcd.conf, but it is extremely odd to have this space-separation for comand line arguments.

I guess one could invoke dhcpcd manually by quoting so that the spaces doesn't make the shell break up the list into multiple args, but still very odd.
Maybe add a comment about the odd syntax here?

Reply:
In dhcpcd they are trying to reuse the code for parsing of the input for both the config file and the command line arguments, resulting in this unfriendly cmd line syntax. I have added a comment on this topic.